src/em/metrics/em_metrics.cpp

```diff
@@ -682,43 +665,43 @@ short em_metrics_t::create_assoc_sta_link_metrics_tlv(unsigned char *buff, mac_a
     short len = 0;
```

Comment:
See above. Making this a `size_t` and having this function return a `size_t` will remove more casts in above functions which require a `short` type for `sz` which could then be changed to a `size_t` 
```suggestion
    size_t len = 0;
```